src/main/java/io/conduit/DestinationStream.java

```diff
@@ -74,8 +77,12 @@ private Destination.Run.Response responseWith(ByteString position) {
     }
 
     private void doWrite(Record record) {
-        task.put(List.of(toSinkRecord(record)));
-        task.flush(emptyMap());
+        SinkRecord sinkRecord = toSinkRecord(record);
+        task.put(List.of(sinkRecord));
+        task.preCommit(Map.of(
```

Comment:
Just curious, what does `preCommit` do?

Reply:
In the Kafka Connect API, it's a hook to be called prior to Kafka Connect actually committing offsets, once certain records have been consumed and written into a sink. preCommit expects to get offsets based on what you've already put into the SinkTask (they even have a comment saying they can track those offsets themselves, not sure why don't they actually do it).

In Snowflake's case, this is how everything fits together:
* When `put` is called, records are put into files (based on batching settings provided), and files are uploaded to a Snowflake stage
* When `preCommit` is called, the connector asks a pipe to move the data from the stage into the configured table.

Different connectors appear to be handling this differently: some do everything in `put` (when a threshold is reached, the data is written), some do it in `preCommit`, and then some in `flush`.

Reply:
> and then some in `flush`

This part made me raise my eyebrow 🤨  Shouldn't we then still call `task.flush(emptyMap());` after `preCommit`?

Reply:
You're welcome!

I was thinking about preCommit and flush too. At the moment, I think it's good to have preCommit only because:
1. Kafka Connect's code itself uses preCommit only, not flush. 
2. By default, preCommit() calls flush()
3. In the KC code, the only reference to flush() is actually preCommit() (its default implementation). Now we can both raise our eyebrows.:D 
4. The connectors I looked at (S3, JDBC, Snowflake, HTTP) implement either only one method, or none at all.